src/instructlab/train/pytorch_train.py

```diff
@@ -291,8 +317,53 @@ def model_generate(user, **kwargs):
     tokenizer.padding_side = "right"
     output_dir = "./training_results"
     per_device_train_batch_size = 1
+
+    if train_style == "quick":
+        do_eval = False
+        evaluation_strategy = "no"
+        load_best_model_at_end = False
+    elif train_style == "full":
+        do_eval = True
+        evaluation_strategy = "epoch"
+        load_best_model_at_end = True
+    # TODO: This might break MacOS
+    fp16 = dtype == "fp16"
+    bf16 = not fp16 and device.type != "mps"
+    training_arguments = TrainingArguments(
+        output_dir=output_dir,
+        num_train_epochs=num_epochs,
+        per_device_train_batch_size=per_device_train_batch_size,
+        fp16=fp16,
+        save_total_limit=2,
+        #  save_strategy = "no",
+        #  load_best_model_at_end=False,
+        # TODO: get EVAL working in order to choose a better epoch
+        load_best_model_at_end=load_best_model_at_end,
+        optim="adamw_torch",
+        evaluation_strategy=evaluation_strategy,
+        do_eval=do_eval,
+        bf16=bf16,
+        use_cpu=False,
+        save_strategy="epoch",
+        report_to="none",
+        torch_compile_backend="aot_eager",
+        torch_compile=True,
```

Comment:
Last time I tried, torch compile did not work for me on AMD ROCm. You also cannot use it work HPUs. The Habana framework / SynapseAI uses a different approach to compile kernels for Intel Gaudi.

Reply:
interesting. This is rly only needed for mps. Will take this into account.

so many corner cases!!!